module/epochs.go

```diff
@@ -36,3 +36,49 @@ type QCContractClient interface {
 	// cluster QC aggregator smart contract for the current epoch.
 	Voted(ctx context.Context) (bool, error)
 }
+
+// DKGContractClient enables interacting with the DKG smart contract. This
+// contract is deployed to the service account as part of a collection of
+// smart contracts that facilitate and manage epoch transitions.
+//
+// Phase parameters may be one of [1,2,3].
+//
+// TODO type for DKGPhase?
+// TODO type for DKGMessage?
+// TODO type for DKGResult? - should be all participant public keys plus group public key
+type DKGContractClient interface {
+
+	// Broadcast broadcasts a message to all other nodes participating in the
+	// DKG for a particular phase. The message is broadcast by submitting a
+	// transaction to the DKG smart contract. An error is returned if the
+	// transaction has failed and should be re-submitted.
+	//
+	// Messages submitted for a phase which has ended will not be accepted and
```

Comment:
The dkg logic does ignore a message that has been received after the deadline, so I'm not sure the contract has to deal with this.